src/platform_impl/linux/x11/event_processor.rs

```diff
@@ -833,6 +849,10 @@ impl<T: 'static> EventProcessor<T> {
                             .focus(xev.event)
                             .expect("Failed to focus input context");
 
+                        let modifiers = ModifiersState::from_x11(&xev.mods);
+
+                        self.device_mod_state.update_state(&modifiers, None);
```

Comment:
This is new part: i'm updating device_mod_state on `XI_Leave` event. Just to be consistent.